src/libraries/System.Private.Xml/src/System/Xml/Core/XmlTextReaderImplHelpers.cs

```diff
@@ -114,6 +114,8 @@ internal int LinePos
                     return charPos - lineStartPos;
                 }
             }
+
+            internal bool StartsWith(string prefix) => chars.AsSpan(charPos).StartsWith(prefix);
```

Comment:
I don't think this is valid. This could end up reading past the populated portion of chars.

Reply:
Hmm, that said, it looks like a bunch of the call sites weren't checking how much of chars was populated.  I'm not sure why that's ok...

Reply:
Ah, I see, those call sites are validating the length is at least the length of the prefix prior to the calls.  That was hidden by GitHub's diff view.